source/Ship.cpp

```diff
@@ -3361,19 +3361,63 @@ double Ship::JumpFuelMissing() const
 // Get the heat level at idle.
 double Ship::IdleHeat() const
 {
-	// This ship's cooling ability:
+	// The Secant Method, closely related to Newton's Method,
+	// is a robust approximation, approaching the zero of any curve
+	// quickly and cheaply. It does not provide infinite accuracy,
+	// but in exchange it is trivial to add more functionality to
+	// the curve it is approximating a root of.
+	// To do so- implement your heat change in ship::DoGeneration()
+	// and then add the same change to Ship::NetIdleHeatAt() below.
+
+	int attempts = 10;
+	double firstGuess = 0.;
+	double firstOutput = NetIdleHeatAt(firstGuess);
+
+	double secondGuess = MaximumHeat();
+	double secondOutput = NetIdleHeatAt(secondGuess);
+
+	double middlingGuess;
+	double middlingOutput;
+
+	if(firstGuess == secondGuess || firstOutput <= secondOutput)
+		return secondOutput > 0. ? numeric_limits<double>::max() : 0;
+
+	// 0.001 here is a magic number which could be anything "sufficiently small".
+	// Normally it's called an Epsilon, a number that approaches but is not zero.
+	while((attempts > 0) && (fabs(firstGuess - secondGuess) > .001))
+	{
+		// Do not allow a negative-heat guess.
+		middlingGuess = max(0., firstGuess - firstOutput * (secondGuess - firstGuess) / (secondOutput - firstOutput));
```

Comment:
Is it possible for secondOutput and firstOutput to be close with the typical values we have?  I'm thinking "no," but you've tested this more than I have. If you think it's a possibility, you should add a safeguard here against a divide-by-something-very-small.